### PR TITLE
[EFIP-6] Execution Layer Rewards Optimization

### PR DESCRIPTION
**Pull request by vvalecha519**

### CODE REVIEW

File: proposals/EFIP-6.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+## Proposal
+1. Add functionality in liquidity pool to mint shares for treasury/EOA
+
+2. Add call to the EtherfiAdmin to mint the shares
+
+3. Upgrade treasury to be able to transfer eeth (only required if we make it payout to treasury)
```
Comment (seongyun-ko) [inline proposals/EFIP-6.md:21]: Shouldn't this proposal about using a sort of Splits contract?

File: proposals/EFIP-6.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Motivation
+
+The benefit of this proposal is that it optimizes gas costs by allowing us to skim execution layer rewards from one address instead of transferring execution rewards from all of our etherfinode contract to the desired address. With the current state of the contracts skimming the eigenpods and the etherficontracts tightly coupled therefore, eth must be transferred from eigenpods and etherfinodes at the same time, whereas with this proposal we could skim the execution layer rewards separatly. The gas cost to skim all execution layer rewards and consensus rewards in batches of 40 is 50K validator / 40 * 8M gas * 8gwei = 80eth. And we can say half of that cost is to withdraw the execution layer rewards. By having 1 address where execution layer rewards flow to from the validators then we only need to make 1 transaction to send all execution layer rewards to the treasury and the cost of that is negliable. Therefore, this solution will save the protocol 40eth per year which is equivalent to 40*$4000 = $160,000.
```
Comment (seongyun-ko) [inline proposals/EFIP-6.md:13]: Currently, the execution layer rewards and MEV feeds of individual validators are being sent to their withdrawal safe (= ether.fi node) contracts and the consensus layer rewards are being sent to their EigenPod contracts. Due to the tightly coupled rewards skimming methods from both contracts, the gas cost is prohibitively large.

By having 1 address where execution layer + MEV rewards flow to from all validators, we only need to make 1 transaction to distribute them to the associate recipients. This solution is expected to save the protocol 40 ETH per year.

File: proposals/EFIP-6.md (hunk 1 above)
Comment (vvalecha519) [reply proposals/EFIP-6.md:21]: i forgot to change this

**Comment by seongyun-ko** [comment]
LGTM

**Pull request merged**